server/sonar-process/src/main/java/org/sonar/process/FileUtils2.java

```diff
@@ -177,4 +193,14 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
       return FileVisitResult.CONTINUE;
     }
   }
+
+  private static final class SizeVisitor extends SimpleFileVisitor<Path> {
+    private long size = 0;
+
+    @Override
+    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+      size += attrs.size();
```

Comment:
value returned by `size()` when file is not a regular file is unspecified (see https://docs.oracle.com/javase/8/docs/api/java/nio/file/attribute/BasicFileAttributes.html#size--), which means is could be `LONG.MAX_VALUE` or negative. Therefor, it's a risk to account for size of non regular files.